packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/dynamic/DynamicMutableRealmObjectTests.kt

```diff
@@ -243,6 +247,23 @@ class DynamicMutableRealmObjectTests {
                                 dynamicSample.set(name, null)
                                 assertEquals(null, dynamicSample.getNullableValue<ByteArray>(name))
                             }
+                            RealmStorageType.ANY -> {
```

Comment:
I think this one is going to be a challenge. We can store `DynamicRealmObject`s, no problem. Getting the object as a `DynamicRealmObject` is a different thing since we don't know whether the link from Core has to be issued as a dynamic or an ordinary object when creating the `RealmAny` instance.

We could extend the dynamic API with `getRealmAny(issueDynamicObjects: Boolean = false)` or something like that, but it is slightly annoying because `getNullableValue` in dynamic API works just fine for all other cases but this one.

Thoughts?